### ISSUE TITLE
helm:Consul helm 0.32.1 / consul 1.10.0 can't inject the service 

### ISSUE THREAD

**Issue by halbornteam** [hashicorp/consul-k8s#640]
### Overview of the Issue

We are on k8s 1.20 and provision the consul-connect for service mesh.  We have deployed the sample application as `static-client` and `static-server` which works properly. However, all of our services can't be started for automatic injection.  

It encounters the error 
>2021-07-16T07:31:52.194Z [ERROR] Unable to get Agent services: error="Unexpected response code: 403 (ACL not found)"
2021-07-16T07:31:53.197Z [ERROR] Unable to get Agent services: error="Unexpected response code: 403 (ACL not found)"
2021-07-16T07:31:54.204Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:31:55.206Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:31:56.210Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:31:57.213Z [INFO]  Unable to find registered services; retrying

We have ACL enabled, and tried to redeploy our service.  The service name matches the service account name. 

### Reproduction Steps

helm -n test-consul install consul hashicorp/consul -f consu.yml

1. When running helm install with the following `values.yml`:
```
  gossipEncryption:
    # secretName is the name of the Kubernetes secret that holds the gossip
    # encryption key. The secret must be in the same namespace that Consul is installed into.
    secretName: "consul-gossip-encryption-key"
    # secretKey is the key within the Kubernetes secret that holds the gossip
    # encryption key.
    secretKey: "key"
  tls:
    # If true, the Helm chart will enable TLS for Consul
    # servers and clients and all consul-k8s components, as well as generate certificate
    # authority (optional) and server and client certificates.
    enabled: true

    enableAutoEncrypt: true
    serverAdditionalDNSSANs:
      - '*.consul'
      - '*.cluster.local'
      - '*.svc.cluster.local'
      - 'localhost.*'

    serverAdditionalIPSANs:
      - '127.0.0.1'
    verify: false
    httpsOnly: true

  acls:
    manageSystemACLs: true
 connect: true
connectInject:
  # True if you want to enable connect injection. Set to "-" to inherit from
  # global.enabled.
  enabled: true
  default: true 
```

1. View error

  --->

### Logs

kubectl -n bots logs -f pod_name consul-connect-inject-init

2021-07-16T07:44:32.721Z [INFO]  Check to ensure a Kubernetes service has been created for this application.
2021-07-16T07:44:33.723Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:34.725Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:35.730Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:36.780Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:37.784Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:38.787Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:39.790Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:40.794Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:41.799Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:42.801Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:42.801Z [INFO]  Check to ensure a Kubernetes service has been created for this application.
2021-07-16T07:44:43.812Z [INFO]  Unable to find registered services; retrying
2021-07-16T07:44:43.812Z [ERROR] Timed out waiting for service registration: error="did not find correct number of services: 0"

--->

### Expected behavior

It init sucessfully and register to consul.

**Comment by wallyhall** [comment]
We're having the same symptom after a Helm upgrade to 1.10.

No pods which were previously running successfully with the sidecar now work - the pod inits all fail with the same error in the connect-inject-init container.

Have tried completely flattening and reinstalling Consul, to no avail.  (Same error.)

**Comment by lkysow** [comment]
@wallyhall can you share the logs from the init containers, the logs from the connect inject deployment, and the output from `kubectl describe service <name>` please.

**Comment by wallyhall** [comment]
> @wallyhall can you share the logs from the init containers, the logs from the connect inject deployment, and the output from `kubectl describe service <name>` please.

Sorry for the late update - I resolved the issue very late (UK local time) last night.  I either overlooked, or the upgrade notes/change log information for Helm on k8s installations didn't make clear that every service in the mesh now requires (as of version 1.10) a Kubernetes `Service` specification.

Adding those to all our services fixed the issue.  (Fortunately I noticed the, imho "very subtle", pale blue which has been added to the Consul/Helm documentation: 

  > Note: As of consul-k8s v0.26.0 and Consul Helm v0.32.0, having a Kubernetes Service is required to run services on the Consul Service Mesh.

cite: https://www.consul.io/docs/k8s/connect#connecting-to-connect-enabled-services )
